gcloud/monitoring/metric.py

```diff
@@ -0,0 +1,241 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Metric Descriptors for the `Google Monitoring API (V3)`_.
+
+.. _Google Monitoring API (V3):
+    https://cloud.google.com/monitoring/api/ref_v3/rest/v3/\
+    projects.metricDescriptors
+"""
+
+# Features intentionally omitted from this first version of the client library:
+#   - Creating and deleting metric descriptors.
+
+import collections
+
+from gcloud.monitoring.label import LabelDescriptor
+
+
+class MetricKind(object):
+    """Allowed values for the `kind of measurement`_.
+
+    .. _kind of measurement:
+        https://cloud.google.com/monitoring/api/ref_v3/rest/v3/\
+        projects.metricDescriptors#MetricKind
+    """
+
+    GAUGE = 'GAUGE'
+    DELTA = 'DELTA'
+    CUMULATIVE = 'CUMULATIVE'
+
+
+class ValueType(object):
+    """Allowed values for the `metric value type`_.
+
+    .. _metric value type:
+        https://cloud.google.com/monitoring/api/ref_v3/rest/v3/\
+        projects.metricDescriptors#ValueType
+    """
+
+    BOOL = 'BOOL'
+    INT64 = 'INT64'
+    DOUBLE = 'DOUBLE'
+    STRING = 'STRING'
+    DISTRIBUTION = 'DISTRIBUTION'
+    MONEY = 'MONEY'
+
+
+class MetricDescriptor(object):
+    """Specification of a metric type and its schema.
+
+    :type name: string
+    :param name:
+        The "resource name" of the metric descriptor. For example:
+        ``"projects/<project_id>/metricDescriptors/<type>"``
+
+    :type type: string
+    :param type:
+        The metric type including a DNS name prefix. For example:
+        ``"compute.googleapis.com/instance/cpu/utilization"``
+
+    :type labels: list of :class:`~gcloud.monitoring.label.LabelDescriptor`
+    :param labels:
+        A sequence of label descriptors specifying the labels used to
+        identify a specific instance of this metric.
+
+    :type metric_kind: string
+    :param metric_kind:
+        The kind of measurement. It must be one of ``MetricKind.GAUGE``,
+        ``MetricKind.DELTA``, or ``MetricKind.CUMULATIVE``.
+        See :class:`MetricKind`.
+
+    :type value_type: string
+    :param value_type:
+        The value type of the metric. It must be one of ``ValueType.BOOL``,
+        ``ValueType.INT64``, ``ValueType.DOUBLE``, ``ValueType.STRING``,
+        ``ValueType.DISTRIBUTION``, or ``ValueType.MONEY``.
+        See :class:`ValueType`.
+
+    :type unit: string
+    :param unit: The unit in which the metric value is reported.
+
+    :type description: string
+    :param description: A detailed description of the metric.
+
+    :type display_name: string
+    :param display_name: A concise name for the metric.
+    """
+
+    def __init__(self, name, type, labels, metric_kind, value_type,
+                 unit, description, display_name):
+        # Allow "type" as a parameter name: pylint: disable=redefined-builtin
```

Comment:
This is another case where we have decided that usability and trueness to the documented API trumps keeping the linter happy. Note that the body does nothing more than initialize attributes.